clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java

```diff
@@ -156,6 +221,97 @@ else if (key.defaultValue == NO_DEFAULT_VALUE)
         return values;
     }
 
+
+    /**
+     * Validate the provided connector configuration values with the configuration definition
+     * @param connectorConfigs the provided connector configuration values
+     * @return List of Config, each Config contains the updated configuration information given
+     * the current configuration values.
+     */
+    public List<Config> validate(Map<String, String> connectorConfigs) {
+        List<String> configsWithNoParent = getConfigsWithNoParent();
+        Map<String, Config> configValues = new HashMap<>();
+        for (String name: configsWithNoParent) {
+            validate(name, connectorConfigs, configValues, name, null);
+        }
+        return new LinkedList<>(configValues.values());
+    }
+
+
+    private List<String> getConfigsWithNoParent() {
+        Set<String> configs = configKeys.keySet();
+        Set<String> configsWithParent = new HashSet<>();
+
+        for (ConfigDef.ConfigKey configKey: configKeys.values()) {
+            List<String> dependents = configKey.dependents;
+            for (String name: dependents) {
+                configsWithParent.add(name);
+            }
+        }
+
+        List<String> configsWithNoParent = new LinkedList<>();
```

Comment:
Can we just use `Set` methods to find the set difference here instead of doing it manually?